src/enemy_ai.rs

```diff
@@ -49,13 +53,43 @@ pub fn set_target_near_player(
         .max_by(f32::total_cmp);
 
     if let Some(max_player_x) = max_player_x {
-        for (e_entity, mut e_trip_point_x, e_transform) in enemies_query.iter_mut() {
+        for (e_entity, mut e_trip_point_x, e_transform, available_attacks) in
+            enemies_query.iter_mut()
+        {
             if let Some(p_transform) = choose_player(&p_transforms, e_transform) {
                 if max_player_x > e_trip_point_x.0 {
                     e_trip_point_x.0 = f32::MIN;
 
-                    let x_offset = rng.gen_range(-ENEMY_TARGET_MAX_OFFSET..ENEMY_TARGET_MAX_OFFSET);
-                    let y_offset = rng.gen_range(-ENEMY_TARGET_MAX_OFFSET..ENEMY_TARGET_MAX_OFFSET);
+                    let mut x_offset =
+                        rng.gen_range(-ENEMY_TARGET_MAX_OFFSET..ENEMY_TARGET_MAX_OFFSET);
+                    let mut y_offset =
+                        rng.gen_range(-ENEMY_TARGET_MAX_OFFSET..ENEMY_TARGET_MAX_OFFSET);
+
+                    let cur_attack = available_attacks.current_attack();
+                    if cur_attack.name.as_str() == "projectile" {
+                        let item = items_assets
+                            .get(&cur_attack.item_handle)
+                            .expect("No item found.");
+
+                        if let ItemKind::Throwable {
+                            lifetime,
+                            throw_velocity,
+                            gravity,
+                            ..
+                        } = item.kind
+                        {
+                            //Change target offset to aim on player
+                            x_offset += throw_velocity.x
+                                * (lifetime * 0.65)
+                                * if p_transform.translation.x > e_transform.translation.x {
+                                    -1.
+                                } else {
+                                    1.
+                                };
+
+                            y_offset += (throw_velocity.y + gravity) * (lifetime * 0.65);
+                        }
```

Comment:
I think this offset calculation is to blame for the slinger sometimes turning around when firing.
The offset was previously used for the melee attack only to add a bit of randomness to where the enemies would chose to stand and attack from, so they wouldnt all pile up directly on top of one another.

I think we should refactor a bit changing the `set_target_near_player` system to `set_move_target_near_player` to clarify intent, and move the projectile targeting code to its own system, which should be run when ranged enemy reaches their move target. likely need to introduce another component for projectile targeting.